pennylane/workflow/execution.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`ExecutionConfig` data class.
+"""
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+from pennylane.interfaces import SUPPORTED_INTERFACES
+from pennylane.gradients import SUPPORTED_GRADIENT_KWARGS
+
+SUPPORTED_GRADIENT_METHODS = [
+    "best",
+    "parameter-shift",
+    "backprop",
+    "finite-diff",
+    "device",
+    "adjoint",
+]
+
+
+@dataclass
+class ExecutionConfig:
+    """
+    A class to configure the execution of a quantum circuit on a device.
+
+    See the Attributes section to learn more about the various configurable options.
+    """
+
+    shots: Optional[Union[int, Tuple[int]]] = None
+    """The number of shots for an execution"""
+
+    gradient_method: Optional[str] = None
+    """The method used to compute the gradient of the quantum circuit being executed"""
+
+    gradient_keyword_arguments: dict = None
+    """Arguments used to control a gradient transform"""
+
+    device_options: dict = None
+    """Various options for the device executing a quantum circuit"""
+
+    framework: str = "jax"
```

Comment:
I'm not sure of the product team's awareness of this, but it was discussed earlier this week in some meeting. I feel like it's a good move, but I'm happy to stick to interface if we feel that it's necessary.

I made the personal decision of avoiding `ml_framework` because I thought it looked rather awkward, but I see both you and @albi3ro proposed it so I will change this (and my other) PR to use `ml_framework`.

Reply:
while this is just an internal implementation for now, I would err on simply keeping `interface` for now pending a decision on porting everything. Partly to avoid confusion, and partly to avoid a situation where we end up with _two_ 'public' conventions.

Reply:
But will defer to @trbromley here